torchvision/prototype/transforms/functional/_color.py

```diff
@@ -183,51 +183,46 @@ def autocontrast(inpt: features.InputTypeJIT) -> features.InputTypeJIT:
         return autocontrast_image_pil(inpt)
 
 
-def _equalize_image_tensor_vec(img: torch.Tensor) -> torch.Tensor:
-    # input img shape should be [N, H, W]
-    shape = img.shape
+def equalize_image_tensor(image: torch.Tensor) -> torch.Tensor:
+    if image.dtype != torch.uint8:
+        raise TypeError(f"Only torch.uint8 image tensors are supported, but found {image.dtype}")
+
+    num_channels, height, width = get_dimensions_image_tensor(image)
+    if num_channels not in (1, 3):
+        raise TypeError(f"Input image tensor can have 1 or 3 channels, but found {num_channels}")
+
+    if image.numel() == 0:
+        return image
+
+    # input image shape should be (*, H, W)
+    shape = image.shape
     # Compute image histogram:
-    flat_img = img.flatten(start_dim=1).to(torch.long)  # -> [N, H * W]
-    hist = flat_img.new_zeros(shape[0], 256)
-    hist.scatter_add_(dim=1, index=flat_img, src=flat_img.new_ones(1).expand_as(flat_img))
+    flat_image = image.flatten(start_dim=-2).to(torch.long)  # -> [*, H * W]
+    hist = flat_image.new_zeros(shape[:-2] + (256,), dtype=torch.int32)
+    hist.scatter_add_(dim=-1, index=flat_image, src=hist.new_ones(1).expand_as(flat_image))
 
     # Compute image cdf
-    chist = hist.cumsum_(dim=1)
+    pixels = flat_image.size(-1)
+    chist = hist.cumsum(dim=-1)
     # Compute steps, where step per channel is nonzero_hist[:-1].sum() // 255
-    # Trick: nonzero_hist[:-1].sum() == chist[idx - 1], where idx = chist.argmax()
-    idx = chist.argmax(dim=1).sub_(1)
-    # If histogram is degenerate (hist of zero image), index is -1
-    neg_idx_mask = idx < 0
-    idx.clamp_(min=0)
-    step = chist.gather(dim=1, index=idx.unsqueeze(1))
-    step[neg_idx_mask] = 0
+    # Trick: nonzero_hist[-1] == hist[chist.argmax()]
+    idx = chist.argmax(dim=-1).unsqueeze_(-1)
+    step = pixels - hist.gather(dim=-1, index=idx)
```

Comment:
nit, for readability if possible :)
```suggestion
    step = nb_pixels - hist.gather(dim=-1, index=idx)
```